go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -0,0 +1,249 @@
+	if !util.PodWantsNetwork(pod) {
+		return nil
+	} else {
+		on, network, err := util.IsNetworkOnPod(pod, bnc.NetInfo)
+		if err == nil && on {
+			nadName := util.GetNadName(network.Namespace, network.Name)
+			key := util.GetSecondaryNetworkLogicalPortName(pod.Namespace, pod.Name, nadName)
+			portInfo, _ := bnc.logicalPortCache.get(key)
+			return portInfo
+		}
+	}
+	return nil
+
+}
+
+// GetInternalCacheEntry4SecondaryNetwork returns the internal cache entry for this object, given an object and its type.
+// This is now used only for pods, which will get their the logical port cache entry.
+func (bnc *BaseNetworkController) GetInternalCacheEntry4SecondaryNetwork(objType reflect.Type, obj interface{}) interface{} {
+	switch objType {
+	case factory.PodType:
+		pod := obj.(*kapi.Pod)
+		return bnc.getPortInfo4SecondaryNetwork(pod)
+	default:
+		return nil
+	}
+}
+
+func (bnc *BaseNetworkController) AddSecondaryNetworkResourceCommon(objType reflect.Type, obj interface{}) error {
+	switch objType {
+	case factory.PodType:
+		pod, ok := obj.(*kapi.Pod)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
+		}
+		return bnc.ensurePod4SecondaryNetwork(pod, true)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+func (bnc *BaseNetworkController) UpdateSecondaryNetworkResourceCommon(objType reflect.Type, oldObj, newObj interface{}, inRetryCache bool) error {
+	switch objType {
+	case factory.PodType:
+		oldPod := oldObj.(*kapi.Pod)
+		newPod := newObj.(*kapi.Pod)
+
+		return bnc.ensurePod4SecondaryNetwork(newPod, inRetryCache || util.PodScheduled(oldPod) != util.PodScheduled(newPod))
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+func (bnc *BaseNetworkController) DeleteSecondaryNetworkResourceCommon(objType reflect.Type, obj, cachedObj interface{}) error {
+	switch objType {
+	case factory.PodType:
+		var portInfo *lpInfo
+		pod := obj.(*kapi.Pod)
+
+		if cachedObj != nil {
+			portInfo = cachedObj.(*lpInfo)
+		}
+		return bnc.removePod4SecondaryNetwork(pod, portInfo)
+
+	default:
+		return fmt.Errorf("object type %s not supported", objType)
+	}
+}
+
+// ensurePod4SecondaryNetwork tries to set up a pod for secondary network. It returns nil on success and error
+// on failure; failure indicates the pod set up should be retried later.
+func (bnc *BaseNetworkController) ensurePod4SecondaryNetwork(pod *kapi.Pod, addPort bool) error {
+
+	// Try unscheduled pods later
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	if !util.PodWantsNetwork(pod) && !addPort {
+		return nil
+	}
+
+	// If a node does node have an assigned hostsubnet don't wait for the logical switch to appear
+	switchName, err := bnc.getExpectedSwitchName(pod)
+	if err != nil {
+		return err
+	}
+
+	if bnc.lsManager.IsNonHostSubnetSwitch(switchName) {
+		return nil
+	}
+
+	on, network, err := util.IsNetworkOnPod(pod, bnc.NetInfo)
```

Comment:
We should probably do `IsNetworkOnPod` before `IsNonHostSubnetSwitch` which holds a mutex. We should be able to very quickly ignore pods that are not attached to this network.